core/output/slack.js

```diff
@@ -100,26 +124,49 @@ renameChannel = function(title, thread) {
 			"channel": teamInfo.channel_id,
 			"name": title
 		};
-		sendMessageWebAPI(body, 'https://slack.com/api/channels.rename', null);
+		addMessageToQueue(teamInfo.team_id, body, 'https://slack.com/api/channels.rename', null);
 	}
 	else {
 		console.debug('slack-> No slack team found!!!');
 	}
 },
 
+getNextMessageId = teamId => {
+	return exports.config.slack_teams[teamId].event_id++;
+}
+
 sendTyping = function(thread) {
 	var slackTeams = exports.config.slack_teams,
-	teamInfo = getChannelIdAndTeamId(thread),
-	socket = sockets[teamInfo.team_id];
+		teamInfo = getChannelIdAndTeamId(thread),
+		teamId = teamInfo.team_id,
+		body = {
+		'type':'typing',
+		'channel': teamInfo.channel_id
+		};
+
+	body.id = getNextMessageId(teamId);
+	body = JSON.stringify(body);
+
+	console.debug("slack-> send typing indication");
+	sendSocketMessage(teamId, body, undefined);
+},
 
+sendSocketMessage = (teamId, body, callback) => {
+	let socket = sockets[teamId];
 	if (socket != null) {
-		var body = {
-			'id': slackTeams[teamInfo.team_id].event_id++,
-			'type':'typing',
-			'channel': teamInfo.channel_id
-		};
-		body = JSON.stringify(body);
-		socket.send(body);
+		socket.send(body, undefined, (response) => {
+			if(response) {
+				console.debug('slack-> Error on socket sending message, error: ' + response);
+			}
+			if (callback) {
+				if (response) {
+					callback(response);
```

Comment:
if `callback` is `undefined`, then `callback(response)` would be the same as `callback()`. This is a redundant if-else.